File: python/3.py
import math


def check(n):
    if n < 2:
        return False
    for i in range(2, int(math.sqrt(n))+1):
        if n % i == 0:
            return False
    return True


p = 0
s = 0
k = 0
q = 0
n = int(input("nhap N: "))
for i in range(1, n+1):
    if (n % i == 0):
        s += 1
        p += i
        if (check(i)):
            k += 1
            q += i
sum = n+p+s-q-k
print("tong bang :", sum)

File: 3.cpp

#include <stdio.h>
#include <math.h>
int check(int n)
{
    if (n == 1)
    {
        return 0;
    }
    for (int i = 2; i <= sqrt(n); i++)
    {
        if (n % i == 0)
        {
            return 0;
        }
    }
    return 1;
}
int main()
{
    int n;
    scanf("%d", &n);
    int s = 0, p = 0;
    int k = 0, q = 0;
    for (int i = 1; i <= n; i++)
    {
        if (n % i == 0)
        {
            p = p + i;
            s++;
            if (check(i) == 1)
            {
                q = q + i;
                k++;
            }
        }
    }
    int tong = n + p + s - q - k;
    printf("%d", tong);
}